### PR TITLE
Refactor [v105] FXIOS-4737 Replace more SDWebImage calls with Kingfisher

### PR DESCRIPTION
**Pull request by nbhasin2**
This replaces direct SDWebimage calls in the following areas with kingfisher

- Top Tabs
- Enhanced Tracking Protection
- Tab Cell
- Pocket 
- Photon Action Sheet

### CODE REVIEW

**Comment by nbhasin2** [comment]
Moving it to draft to fix some tests

**Comment by nbhasin2** [comment]
This is good to review, the tests were passing except for the export command and in the last commit, I fixed swift lint rules.

File: Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift [hunk 1]
```diff
@@ -333,11 +333,17 @@ class EnhancedTrackingProtectionMenuVC: UIViewController {
     }
 
     private func updateViewDetails() {
+        heroImage.image = UIImage(named: "defaultFavicon")!
```
Comment (lmarceau) [inline Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift:336]: nit, doesn't have to be forced
```suggestion
        heroImage.image = UIImage(named: "defaultFavicon")
```

File: Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift [hunk 2]
```diff
@@ -333,11 +333,17 @@ class EnhancedTrackingProtectionMenuVC: UIViewController {
...
         if let favIconURL = viewModel.favIcon {
-            heroImage.sd_setImage(with: favIconURL, placeholderImage: UIImage(named: "defaultFavicon"), options: [], completed: nil)
-        } else {
-            heroImage.image = UIImage(named: "defaultFavicon")!
-            heroImage.tintColor = UIColor.theme.etpMenu.defaultImageTints
+            ImageLoadingHandler.shared.getImageFromCacheOrDownload(with: favIconURL,
+                                                                   limit: ImageLoadingConstants.NoLimitImageSize) { image, error in
+                guard error == nil, let image = image else {
+                    return
+                }
```
Comment (lmarceau) [inline Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift:343]: nit
```suggestion
                guard error == nil, let image = image else { return }
```

File: Client/Frontend/Browser/TabCell.swift [hunk 3]
```diff
@@ -176,11 +176,16 @@ class TabCell: UICollectionViewCell, TabTrayCell, ReusableCell {
...
         if let favIcon = tab.displayFavicon, let url = URL(string: favIcon.url) {
-            favicon.sd_setImage(with: url, placeholderImage: UIImage(named: "defaultFavicon"), options: [], completed: nil)
-        } else {
-            favicon.image = UIImage(named: "defaultFavicon")
-            favicon.tintColor = UIColor.theme.tabTray.faviconTint
+            ImageLoadingHandler.shared.getImageFromCacheOrDownload(with: url,
+                                                                   limit: ImageLoadingConstants.NoLimitImageSize) { image, error in
+                guard error == nil, let image = image else {
+                    return
+                }
```
Comment (lmarceau) [inline Client/Frontend/Browser/TabCell.swift:186]: nit
```suggestion
                guard error == nil, let image = image else { return }
```

File: Client/Frontend/Browser/TopTabsViews.swift [hunk 4]
```diff
@@ -116,13 +116,19 @@ class TopTabCell: UICollectionViewCell, NotificationThemeable, TabTrayCell, Reus
...
-        } else {
-            favicon.image = UIImage(named: "defaultFavicon")
-            favicon.tintColor = UIColor.theme.tabTray.faviconTint
-            favicon.contentMode = .scaleAspectFit
-            favicon.backgroundColor = .clear
+            ImageLoadingHandler.shared.getImageFromCacheOrDownload(with: url,
+                                                                   limit: ImageLoadingConstants.NoLimitImageSize) { image, error in
+                guard error == nil, let image = image else {
+                    return
+                }
```
Comment (lmarceau) [inline Client/Frontend/Browser/TopTabsViews.swift:129]: nit
```suggestion
                guard error == nil, let image = image else { return }
```

File: Client/Frontend/Home/Pocket/PocketStandardCell.swift [hunk 5]
```diff
@@ -140,7 +141,14 @@ class PocketStandardCell: UICollectionViewCell, ReusableCell {
         descriptionLabel.text = viewModel.description
         accessibilityLabel = viewModel.accessibilityLabel
 
-        heroImageView.sd_setImage(with: viewModel.imageURL)
+        ImageLoadingHandler.shared.getImageFromCacheOrDownload(with: viewModel.imageURL,
+                                                               limit: ImageLoadingConstants.NoLimitImageSize) { image, error in
+            guard error == nil, let image = image else {
+                return
+            }
```
Comment (lmarceau) [inline Client/Frontend/Home/Pocket/PocketStandardCell.swift:148]: nit
```suggestion
            guard error == nil, let image = image else { return }
```

File: Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift [hunk 6]
```diff
@@ -333,11 +333,17 @@ class EnhancedTrackingProtectionMenuVC: UIViewController {
...
+
+                self.heroImage.image = image
```
Comment (lmarceau) [inline Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift:345]: For `self` in `getImageFromCacheOrDownload` calls, do we need to weakify `self` here? (I won't make the comment in each function haha, you get the point).

**Comment by lmarceau** [comment]
Build was green [here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/2184bf3d-bbb4-4ce4-8253-6287bbaaeb0f), before nits.

Screen Shot 2022-08-10 at 4 10 52 PM

**Review by lmarceau — APPROVED** [review]

File: Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift (hunk 6 above)
Comment (nbhasin2) [reply Client/Frontend/Browser/EnhancedTrackingProtection/EnhancedTrackingProtectionVC.swift:343]: Ah good catch, it should be weak

**Pull request merged**